bpfd-operator/controllers/bpfd-agent/kprobe-program.go

```diff
@@ -213,6 +219,13 @@ func (r *KprobeProgramReconciler) reconcileBpfdProgram(ctx context.Context,
 		}
 
 		// otherwise load it
+		bytecode, err := bpfdagentinternal.GetBytecode(r.Client, bytecodeSelector)
+		if err != nil {
+			return bpfdiov1alpha1.BpfProgCondBytecodeSelectorError, fmt.Errorf("failed to process bytecode selector: %v", err)
```

Comment:
Hrm I know we might not always need the bytecode.. but can you just call this function and buildKprobeLoadRequest once up near the entry of reconcileBpfdProgram? I don't like having the code block duplicated, and IMO this whole thing should ALWAYS fail if we don't have access to the bytecode

If you really don't want to do that please just put it in an in-line anonymous and call it when needed.

Reply:
If we just call it once at the top and return if we get an error, then we can't delete the bpfProgram objects that are created.  I could call it at the top, remember whether there's an error, and then allow deletes, but only do loads if it was successful.

Reply:
> If we just call it once at the top and return if we get an error, then we can't delete the bpfProgram objects that are created.

Hrm yeah that's an issue :( 

see below comment, it allows us to deduplicate that code block without adding a new function def

Reply:
I still need to make this change.

Reply:
Done.